sphinx/util/inspect.py

```diff
@@ -501,6 +503,63 @@ def __repr__(self) -> str:
         return self.value
 
 
+class TypeAliasForwardRef:
+    """Pseudo typing class for autodoc_type_aliases.
+
+    This avoids the error on evaluating the type inside `get_type_hints()`.
+    """
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def __call__(self) -> None:
+        # Dummy method to imitate special typing classes
+        pass
+
+    def __eq__(self, other: Any) -> bool:
+        return self.name == other
+
+
+class TypeAliasNamespace(Dict[str, Any]):
```

Comment:
It might be better to separate a "namespace" and a "module". They are very similar, but different things.

* Namespace
  * Accessed via `__getitem__` API
  * Not having a "current module"
  * Returns a module or forwardref.
  * Raises KeyError if not found
* Module
  * Accessed via `__getattr__` API
  * Has its "current module" name.
  * Returns a sub-module or forwardref, or real object
  * Never raises KeyError.